bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java

```diff
@@ -0,0 +1,131 @@
+
+public class SearchResultsStorefrontContextImpl extends AbstractCommerceStorefrontContext implements SearchResultsStorefrontContext {
+
+    private final SearchResultsSet searchResultsSet;
+
+    public SearchResultsStorefrontContextImpl(SearchResultsSet searchResultsSet) {
+        this.searchResultsSet = searchResultsSet;
+    }
+
+    @Override
+    public String getSearchUnitId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getSearchRequestId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public List<SearchResultProduct> getProducts() {
+        return searchResultsSet.getProductListItems().stream()
+            .map(p -> new SearchResultProduct() {
+
+                @Override
+                public String getName() {
+                    return p.getTitle();
+                }
+
+                @Override
+                public String getSku() {
+                    return p.getSKU();
+                }
+
+                @Override
+                public String getUrl() {
+                    return p.getURL();
+                }
+
+                @Override
+                public String getImageUrl() {
+                    return p.getImageURL();
+                }
+
+                @Override
+                public Double getPrice() {
+                    return p.getPriceRange().getFinalPrice();
+                }
+
+                @Override
+                public int getRank() {
+                    return 0;
+                }
+            }).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<SearchResultCategory> getCategories() {
+        SearchAggregation searchAggregationOptions = searchResultsSet
+            .getSearchAggregations().stream()
+            .filter(a -> a.getIdentifier().equals("category_id"))
+            .findFirst()
+            .orElse(null);
+
+        if (searchAggregationOptions != null) {
+            return searchAggregationOptions.getOptions().stream()
+                .map(o -> new SearchResultCategory() {
+
+                    @Override
+                    public String getName() {
+                        return o.getDisplayLabel();
+                    }
+
+                    @Override
+                    public String getUrl() {
+                        return StringUtils.EMPTY;
+                    }
+
+                    @Override
+                    public int getRank() {
+                        return 0;
+                    }
+                }).collect(Collectors.toList());
+        }
+
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<SearchResultSuggestion> getSuggestions() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public int getPage() {
+        return searchResultsSet.getSearchOptions().getCurrentPage();
+    }
+
+    @Override
+    public int getPerPage() {
+        return searchResultsSet.getSearchOptions().getPageSize();
+    }
+
```

Comment:
For completeness of the interface, could you add the `facets` attribute as well?

https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/searchResults.ts#L18

Reply:
and return `null` for in the implementation?

Reply:
For now yes. We can fully utilise it, when we migrate to Adobe Commerce Live Search.